src/mfs.jl

```diff
@@ -0,0 +1,163 @@
+    using the formula for the n-sphere.
+"""
+
+
+function random_point_on_sphere(radius, n)
+    angles = rand(n-1) * 2π  
+
+    coordinates = [radius * prod(sin.(angles[1:i-1])) * cos(angles[i]) for i in 1:n-1]
+    push!(coordinates, radius * prod(sin.(angles)))  
+    
+    return coordinates
+end
+
+""" 
+    random_pertubation(mapper, X, search_area, dim,  Ntries=10000)
+
+Input:
+    mapper: AttractorMapper
+    X: initial point to test
+    search_area: array of two points defining the search area
+    dim: dimension of the system
+    Ntries: number of random pertubations to try, default = 10000
+
+Output:
+    best_shock: vector of the best shock found
+    best_dist: norm of the best shock found
+
+Description:
+    This function generates a random pertubation of the initial point "X" within specified "search_area" and checks if it is in the same basin of attraction.
+    It does so by generating a random vector of length dim and then adding it to X.
+    If the pertubation is not in the same basin of attraction, it calculates the norm of the pertubation and compares it to the best pertubation found so far.
+    If the norm is smaller, it updates the best pertubation found so far.
+    It repeats this process Ntries times and returns the best pertubation found.
+
+"""
+
+function random_pertubation(mapper::AttractorMapper, X, search_area, dim,  Ntries=10000)
+    best_dist = Inf
+    best_shock = nothing
+    Xsp = StateSpaceSet([X])
+    for _ in 1:Ntries
+        u0 = search_area[1] .+ rand(dim) .* (search_area[2] .- search_area[1])
+        shock = X + u0
+        if !(check_basin(mapper, shock, X))
+            dist = LinearAlgebra.norm(u0)
+            if dist < best_dist
+                best_dist = dist
+                best_shock = u0
+            end
+        end
+    end
+    return best_shock, best_dist
+end
+
+
+""" 
+    mfs_brute_force(mapper, X, best_shock, best_dist, dim,  Ntries=10000)
+
+Input:
+    mapper: AttractorMapper
+    X: initial point to test
+    best_shock: vector of the best shock found so far
+    best_dist: norm of the best shock found so far
+    dim: dimension of the system
+    Ntries: number of random pertubations to try, default = 10000
+
+Output:
+    best_shock: vector of the best shock found
+    best_dist: norm of the best shock found
+
+Description:
+    This function works on the results obtained by random_pertubation(). It starts from the best shock found so far and tries to find a better one 
+    by continuously reducing the radius of the sphere on the surface of which it generates random pertubations. If pertubation with the same basin of attraction is found,
+    it updates the best shock found so far and reduces the radius of the sphere. 
+    It repeats this process Ntries times and returns the best pertubation found.
+"""
+
+
+function mfs_brute_force(mapper::AttractorMapper, X, best_shock, best_dist, dim,  Ntries=10000)
```

Comment:
This function should utilize a new Type created for the `minimal_fatal_shock` main function (see comment below). That type must contain all information related with the brute force algorithm , like the number of tries used in the initial random perturbation search, as well as the number of tries used in the "minimization" step of successively reducing the hypersphere radius.